### PR TITLE
fix: [CDS-99267]: Update toggle pill color for dark and light mode

### PR DESCRIPTION
**Pull request by sarthak-setia**
You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- StyleLint: `retrigger stylelint`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- ImageSnapshot `retrigger ImageSnapshot`

Screenshot 2024-10-01 at 12 41 57 PM

Screenshot 2024-10-01 at 12 42 21 PM

Screenshot 2024-10-01 at 12 42 03 PM

Screenshot 2024-10-01 at 12 42 17 PM

### CODE REVIEW

**Comment by bot2-harness** [comment]
Image snapshots failed please check https://console.cloud.google.com/storage/browser/ui-team-bucket/uicore-2656

**Comment by bot2-harness** [comment]
Image snapshots failed please check https://console.cloud.google.com/storage/browser/ui-team-bucket/uicore-2667

File: packages/uicore/src/components/Toggle/Toggle.css [hunk 1]
```diff
@@ -9,8 +9,8 @@
   --toggle-background-color: var(--white);
   --toggle-border-color: #c7c9d9; /* grey-200 too light and grey-250/300 too dark */
   --toggle-pill-size: var(--spacing-4);
-  --toggle-pill-color: var(--white);
-  --toggle-pill-shadow: 0 1px 2px rgba(40, 41, 61, 0.2), 0 2px 4px rgba(96, 97, 112, 0.2);
+  --toggle-pill-color: #c7c9d9; /* Needs to be consistent in light and dark mode */
```
Comment (MondalSayantan) [inline packages/uicore/src/components/Toggle/Toggle.css:12]: We have a variable for this color:  `--toggle-border-color`. Let's use that

**Comment by MondalSayantan** [comment]
Please add screenshots for light and dark mode

**Comment by bot2-harness** [comment]
Image snapshots failed please check https://console.cloud.google.com/storage/browser/ui-team-bucket/uicore-2674

**Comment by MondalSayantan** [comment]
Please bump the uicore package

**Comment by bot2-harness** [comment]
Image snapshots failed please check https://console.cloud.google.com/storage/browser/ui-team-bucket/uicore-2701

**Comment by bot2-harness** [comment]
Image snapshots failed please check https://console.cloud.google.com/storage/browser/ui-team-bucket/uicore-2723

**Review by MondalSayantan — APPROVED** [review]

**Review by avinashmadhwani02 — APPROVED** [review]

**Pull request merged**